libvcx/src/aries/handlers/proof_presentation/prover/prover.rs

```diff
@@ -136,9 +133,12 @@ impl Prover {
                 .filter_map(|(_, message)| {
                     match message {
                         A2AMessage::PresentationRequest(presentation_request) => {
-                            Some(presentation_request)
+                            match (presentation_request.request_presentations_attach.content().ok(), match_name) {
+                                (Some(content), Some(match_name)) => _filter_by_name(&content, match_name, presentation_request),
+                                _ => Some(presentation_request)
```

Comment:
The behaviour is bit unexpected:
- if I specify match name and the attachmed is not valid json, the proof request is still considered as matched

Now previously `match_name` was never used here, content was not inspected and it was seemingly up to the caller to inspect the content and deal with it.

I understand this code is useful for the test you are writing. But there's no external usecase for `match_name` and the behaviour is surprising. 

Here's my proposal though -  how about we extract this filtering logic as sort of util function which you could do in your test (fetch all the presentation requests messages, then filter the one you need yourself) and perhaps even expose as C-API. 
That would also be more maintainable way going into future - adding different filtering strategies would be as simple as adding a new function (which again, each could be exposed on C-API layer too)

So we would now remove `vcx_v2_disclosed_proof_get_requests` and keep the old  `vcx_vdisclosed_proof_get_requests`
Instead we would add
 - `vcx_filter_proof_requests_by_name`
and in future we could add more, like 
- ``vcx_filter_proof_requests_by_issuer` etc.

Reply:
I like the idea, but it is probably the simplest and most easily extensible option we have, and those benefits are worth it. I will expose the first `vcx_filter_*` function as part of this PR and use it in the tests.

Reply:
cool! 👍

Reply:
However, "if I specify match name and the attachmed is not valid json, the proof request is still considered as matched" is incorrect - `_filter_by_name` will return None in that case